src/nodes/chunk_dispatch/chunk_insert_state.c

```diff
@@ -585,6 +585,12 @@ ts_chunk_insert_state_create(const Chunk *chunk, ChunkDispatch *dispatch)
 	OnConflictAction onconflict_action = chunk_dispatch_get_on_conflict_action(dispatch);
 	ResultRelInfo *relinfo;
 
+	if (chunk->fd.compressed_chunk_id != INVALID_CHUNK_ID && !ts_guc_enable_dml_decompression)
```

Comment:
I think this also needs to get pushed down into decompress_batches_for_insert and block only if there are unique or primary constraints on the chunk.

Reply:
Fixed!